doc/internals.rst

```diff
@@ -129,9 +129,77 @@ Back in an interactive IPython session, we can use these properties:
 
 The intent here is that libraries that extend xarray could add such an accessor
 to implement subclass specific functionality rather than using actual subclasses
-or patching in a large number of domain specific methods. For further reading
-on ways to write new accessors and the philosophy behind the approach, see
-:issue:`1080`.
+or patching in a large number of domain specific methods.
+
+Parametrizing an accessor is possible by defining ``__call__``. For
+example, we could use this to reimplement the API of the
+:py:func:`DataArray.weighted` function:
+
+.. ipython::
+    :okwarning:
+
+    In [1]: @xr.register_dataarray_accessor("weighted")
+       ...: class WeightedAccessor:
+       ...:     def __init__(self, xarray_obj):
+       ...:         self._obj = xarray_obj
+       ...:         self._weights = np.ones_like(xarray_obj) / xarray_obj.size
+       ...:
+       ...:     def __call__(self, weights):
+       ...:         self._weights = weights
+       ...:         return self
```

Comment:
This approach seems potentially a little error prone because it involves mutation. Could we possibly mention it less prominently? or maybe not at all?

Reply:
That's true, but I don't really understand why this would be an issue if you only ever call it with `da.accessor()` or `da.accessor(data)`. We could also do
```python
class Accessor:
    def __init__(self, obj, data=None):
        self._obj = obj
        self._data = data

    def __call__(self, data=None):
        return type(self)(self._obj, data)
```
which does not rely on the internal state but is a little bit more complex.